pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -195,6 +196,7 @@ class RichProgressBar(ProgressBarBase):
 
     Args:
         refresh_rate_per_second: the number of updates per second. If refresh_rate is 0, progress bar is disabled.
+        display_every_n_epochs: Set to a non-negative integer to display progress bar every n epochs.
```

Comment:
should it be restricted to just -1?

Reply:
What do you mean by `restricted to just -1`?

Reply:
I meant.. currently setting it to -2, -100 will show it on every epochs. so should we just make it -1? else raise a Misconfig.. the reason being, we have certain other flags across codebase that takes just `-1` as an indication for no restriction.

Reply:
No, it will currently [raise](https://github.com/PyTorchLightning/pytorch-lightning/blob/support%2Fdisplay_every_n_epoch/pytorch_lightning/callbacks/progress/rich_progress.py#L219) a Misconfiguration exception.

Reply:
okay.. sorry my bad.. it's non-negative.. 🤦‍♂️

Reply:
"display" is misleading as a name imo. I wasn't sure what you mean by it, and I tested it out. It actually means you want to keep the last line of the progress bar in the stdout print at the end of an epoch, if I interpret correctly. 

Have we considered/discussed the name to be something else? Why do you think it is important to set the frequency to a different value than 0 or 1? 

`RichProgressBar(leave=True|False)`
`RichProgressBar(leave_every_epoch=True|False)`

Reply:
how about `bar_every_epoch`? and set it to true false rather than providing an actual value?

Reply:
But if I understand (hopefully), the question is not about whether the user wants to see a bar or not (it will be there the entire time). The question is whether once it is full, should it delete the full bar and reset completely or should it make a new line to start a new bar (and leave the old one behind).

Reply:
I am okay with boolean values.

`display_bar_every_epoch` or `bar_every_epoch`?

Tbh, `display_every_epoch` sounds good to me. Because you are technically displaying each epoch progress bar to user on the terminal.

Reply:
`progress_every_epoch`?

Reply:
Or should we go with `reset_every_epoch` with True as default.

Reply:
I don't have the full context. But one clarification:

The progress bar still appears and gets updated during the epoch and this feature is about whether it stays printed on the terminal afterward?

If that's correct, I'm in favor of naming it `leave` which matches exactly the `tqdm`
 API

Reply:
so if I read it correctly, keeps showing last n bars from particular epochs, so `preserve_n_last_epochs_bars`?

Reply:
> I think we could be more intuitive about it and we don't need to match the tqdm api

I agree, but I think it is easier for people if we provide the same interface for the same functionality from different backends.

Reply:
@tchaton @SeanNaren I need two more votes, before going ahead with it. Let's make it quick!